PcdcAnalysisTools/blueprint/routes/views/counts/__init__.py

```diff
@@ -47,7 +47,8 @@ def get_result():
     counts = get_counts_list(data, args)
     # Cache is invalidated if the service is restarted or if the reset_count_cache endpoint is used
     capp.logger.info("CACHING COUNTS VALUE - " + json.dumps(counts))
-    capp.config["cache"]["counts"] = counts
+    if(capp.mock_data != 'True'):
```

Comment:
If the image here is initialized correctly, I don't think you need to add this if statement to skip this.
<img width="536" alt="image" src="https://github.com/user-attachments/assets/33d333a5-7d91-471e-a2e2-0111839e6b9e">

Reply:
In lines 27-30 of the get_result function in .../views/counts/__init__.py:
`    if cache:
        if "counts" in cache and cache["counts"] is not None:
            capp.logger.info("RETRIEVING CACHE COUNTS VALUE - " + json.dumps(cache["counts"]))
            return flask.jsonify(cache["counts"])
`
If I get rid of the if statement to skip the cache, then two of the tests return errors because instead of accessing their own data they access the same data as the first counts test, and then there is an assertion error since each test is supposed to be accessing different sets of data.

Reply:
Then you want to reset the cache before each tests instead. The if statement in the application code is a little weird for this. 
You can do something like this - https://flask.palletsprojects.com/en/1.1.x/testing/#accessing-and-modifying-sessions
Or something like this: https://stackoverflow.com/questions/73865462/reset-global-variables-between-two-tests-on-pytest#answer-73872448

I am sure there are other solutions, these are just the first two I found with a quick google search.

Reply:
I added a test fixture to reset the cache and removed the conditional from the application code, all the tests are now passing